torchgeo/datasets/rioxarray.py

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""In-memory geographical xarray.DataArray."""
+
+import glob
+import os
+import re
+import sys
+from datetime import datetime
+from typing import Any, Callable, Optional, cast
+
+import netCDF4  # noqa: F401
+import numpy as np
+import torch
+import xarray as xr
+from rasterio.crs import CRS
+from rtree.index import Index, Property
+
+from .geo import GeoDataset
+from .utils import BoundingBox
+
+
+class RioXarrayDataset(GeoDataset):
+    """Wrapper for geographical datasets stored as Xarray Datasets.
+
+    Relies on rioxarray.
+    """
+
+    filename_glob = "*"
+    filename_regex = ".*"
+
+    def __init__(
+        self,
+        root: str,
+        data_variables: list[str],
+        crs: Optional[CRS] = None,
+        res: Optional[float] = None,
+        transforms: Optional[Callable[[dict[str, Any]], dict[str, Any]]] = None,
+    ) -> None:
+        """Initialize a new Dataset instance.
+
+        Args:
+            root: directory with nc files
+            data_variables: data variables that should be gathered from the xr_datasets
```

Comment:
Is `xr_datasets` a variable? Or do you mean to say xarray datasets?

Reply:
sorry meant to say xarray datasets